### PR TITLE
flash_attention_helper mitigation: pass contiguous inputs

### PR DESCRIPTION
**Pull request by cpuhrsch**
There appears to be a transient issue with respect to non-contiguous inputs in flash_attn and thus we're passing contiguous inputs to mitigate it.

### CODE REVIEW

File: aten/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp [hunk 1]
```diff
@@ -296,18 +296,26 @@ Tensor flash_attention_helper(
...
+    auto i2 = qkv_buffer_reshaped[2];
+
+    TORCH_CHECK(i0.is_contiguous());
+    TORCH_CHECK(i1.is_contiguous());
+    TORCH_CHECK(i2.is_contiguous());
 
     // If we are passing in query, key, value all the same tensors then we have
     // packed them into one tensor and need to slice for flash attention
     Tensor atten_buffer = at::_flash_scaled_dot_product_attention(
-        qkv_buffer_reshaped.index({at::indexing::Slice(), 0}),
```
Comment (drisspg) [inline aten/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp:308]: I just added the `qkv_buffer_reshaped.index({at::indexing::Slice(), 0}).contiguous()` for the next three and that seemed to fix as well. Not sure if that is less perf but felt like fewer dispatch calls ... premature optimization is the way

Comment (cpuhrsch) [reply aten/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp:308]: That'll cause one call to `contiguous` for each Tensor, which means 3 kernel launches. I don't think these memcpys take a lot of GPU time (didn't look at a trace), but if we can do it with one call to contiguous I suspect it's more efficient.

Comment (drisspg) [reply aten/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp:308]: even more interestingly, at least to me, calling contiguous on the individual q,k,v slices fixes the repro however there are still cuda memory errors for padding patterns. This method of transposing then making the entire packed project contiguous appears to work for the entire sweep.

**Review by drisspg — APPROVED** [review]
Nit: I think I would call i0, i1, i2 -> q,k,v but doesn't matter

**Pull request closed without merging**